filament/src/Renderer.cpp

```diff
@@ -235,6 +235,40 @@ void FRenderer::renderJob(ArenaScope& arena, FView* view) {
     recordHighWatermark(commands);
 }
 
+void FRenderer::mirrorFrame(FSwapChain* dstSwapChain, Viewport const& dstViewport, Viewport const& srcViewport,
+                            MirrorFrameFlag flags) {
+    SYSTRACE_CALL();
+
+    assert(mSwapChain);
+    assert(mSwapChain->isReadable());
+    assert(dstSwapChain);
+    FEngine& engine = getEngine();
+    FEngine::DriverApi& driver = engine.getDriverApi();
+
+    const Handle<HwRenderTarget> viewRenderTarget = getRenderTarget();
+
+    // Set the current swap chain as the read surface, and the destination
+    // swap chain as the draw surface so that blitting between default render
+    // targets results in a frame copy from the current frame to the
+    // destination.
+    driver.makeCurrent(dstSwapChain->getHwHandle(), mSwapChain->getHwHandle());
+
+    driver.blit(TargetBufferFlags::COLOR,
```

Comment:
Check that the source SwapChain has the IS_READABLE flag, so we can ensure clients use this properly. I would use an assert() at least.

Reply:
it was earlier in the function.  Moved it right before the blit and added a comment.